integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -353,3 +355,38 @@ async fn test_multichain_reshare_with_lake_congestion() -> anyhow::Result<()> {
     })
     .await
 }
+
+#[test(tokio::test)]
+async fn test_multichain_update_contract() -> anyhow::Result<()> {
+    let config = MultichainConfig::default();
+    with_multichain_nodes(config.clone(), |ctx| {
+        Box::pin(async move {
+            // Get into running state and produce a singular signature.
+            let state = wait_for::running_mpc(&ctx, Some(0)).await?;
+            wait_for::has_at_least_mine_triples(&ctx, 2).await?;
+            wait_for::has_at_least_mine_presignatures(&ctx, 1).await?;
+            actions::single_payload_signature_production(&ctx, &state).await?;
+
+            // Perform update to the contract and see that the nodes are still properly running and picking
+            // up the new contract by first upgrading the contract, then trying to generate a new signature.
+            let id = ctx.propose_update_contract_default().await;
```

Comment:
Ideally, we should check for contract hash and ensure it changed to the one we were proposing. Signature production can work with the old one.

Reply:
Checking whether or not these values have been changed should just be reserved for the unit tests which we test against already. Integration tests are more about if the change makes it so that the nodes aren't running anymore. That being said, this test doesn't really do that quite yet. We should eventually add something like an invalid contract upgrade and see whether or not the nodes behave as expected